staging/cse/windows/provisioningscripts/cleanupnetwork.ps1

```diff
@@ -1,37 +1,42 @@
 
 $global:NetworkMode = "L2Bridge"
 $global:NetworkPlugin = $Global:ClusterConfiguration.Cni.Name
 $global:HNSModule = "c:\k\hns.v2.psm1"
 
 ipmo $global:HNSModule
 
-$networkname = $global:NetworkMode.ToLower()
-if ($global:NetworkPlugin -eq "azure") {
-    $networkname = "azure"
-}
+if ($CleanupHnsNetwork) { # CleanupHnsNetwork is legacy code
+    $networkname = $global:NetworkMode.ToLower()
+    if ($global:NetworkPlugin -eq "azure") {
+        $networkname = "azure"
+    }
 
-$hnsNetwork = Get-HnsNetwork | ? Name -EQ $networkname
-if ($hnsNetwork) {
-    # Cleanup all containers
-    Write-Host "Cleaning up containers"
-    ctr.exe -n k8s.io c ls -q | ForEach-Object { ctr -n k8s.io tasks kill $_ }
-    ctr.exe -n k8s.io c ls -q | ForEach-Object { ctr -n k8s.io c rm $_ }
-    
-    Write-Host "Cleaning up persisted HNS policy lists"
-    # Initially a workaround for https://github.com/kubernetes/kubernetes/pull/68923 in < 1.14,
-    # and https://github.com/kubernetes/kubernetes/pull/78612 for <= 1.15
-    #
-    # October patch 10.0.17763.1554 introduced a breaking change 
-    # which requires the hns policy list to be removed before network if it gets into a bad state
-    # See https://github.com/Azure/aks-engine/pull/3956#issuecomment-720797433 for more info
-    # Kubeproxy doesn't fail becuase errors are not handled: 
-    # https://github.com/delulu/kubernetes/blob/524de768bb64b7adff76792ca3bf0f0ece1e849f/pkg/proxy/winkernel/proxier.go#L532
-    Get-HnsPolicyList | Remove-HnsPolicyList
+    $hnsNetwork = Get-HnsNetwork | ? Name -EQ $networkname
+    if ($hnsNetwork) {
+        # Cleanup all containers
+        Write-Host "Cleaning up containers"
+        ctr.exe -n k8s.io c ls -q | ForEach-Object { ctr -n k8s.io tasks kill $_ }
+        ctr.exe -n k8s.io c ls -q | ForEach-Object { ctr -n k8s.io c rm $_ }
+        
+        Write-Host "Cleaning up persisted HNS policy lists"
+        # Initially a workaround for https://github.com/kubernetes/kubernetes/pull/68923 in < 1.14,
+        # and https://github.com/kubernetes/kubernetes/pull/78612 for <= 1.15
+        #
+        # October patch 10.0.17763.1554 introduced a breaking change 
+        # which requires the hns policy list to be removed before network if it gets into a bad state
+        # See https://github.com/Azure/aks-engine/pull/3956#issuecomment-720797433 for more info
+        # Kubeproxy doesn't fail becuase errors are not handled: 
+        # https://github.com/delulu/kubernetes/blob/524de768bb64b7adff76792ca3bf0f0ece1e849f/pkg/proxy/winkernel/proxier.go#L532
+        Get-HnsPolicyList | Remove-HnsPolicyList
 
-    Write-Host "Cleaning up old HNS network found"
-    Remove-HnsNetwork $hnsNetwork
-    Start-Sleep 10
+        Write-Host "Cleaning up old HNS network found"
+        Remove-HnsNetwork $hnsNetwork
+        Start-Sleep 10
+    }
+} else {
+    Write-Host "Skipping network legacy code: Remove-HnsNetwork"
 }
 
```

Comment:
Could we still remove those json files of azurecni if we do not remove hnsnetwork?

Reply:
The cleanup process for `HNS Network` and `CNI data` should be consistent.